Camel 3.17 - whats new blog (#846)
apache/camel-website: content/blog/2022/05/camel317-whatsnew/index.md

---
title: "Apache Camel 3.17 What's New"

> date: 2022-03-23

Reviewer: This date might be updated ?

authors: [davsclaus]
categories: ["Releases"]
preview: Details of what we have done in the Camel 3.17 release.
---

Apache Camel 3.17 has just been TODO: released(/blog/2022/05/RELEASE-3.17.0/).

This release introduces a set of new features and noticeable improvements that we will cover in this blog post.

### Java 17 (runtime) support

> This is the first release where we have official Java 17 support. The support is runtime only,

Reviewer: Really easy to remember. Java 17 support is introduced in Camel 3.17 :)

Author: Ah of course, lets add a sentence about this, you are welcome to add that.

meaning that we do not special support for new Java 17 language features such as Java records.

### Camel Main

> When running Camel standalone via `camel-main` then we have improved the startup summary
> to list used configurations for auto-configuration and property placeholders.

Reviewer 2: ```suggestion
We have improved the startup summary to list used configurations for auto-configuration and property placeholders
when running Camel standalone via `camel-main`.
```

The information displayed will show the source for each configuration; in other words

> you can exactly see if the configuration is from application.properties, OS environment variable,

Reviewer 2: ```suggestion
you can exactly see if the configuration is from `application.properties`, OS environment variable,
```

command line argument, or elsewhere.

### Camel JBang (Camel CLI)

In this release we have focused the most new development on the effort for Camel JBang.
This is a new way of getting started and using Apache Camel.

> Users that are familiar with Camel K, and its "easy of use", then we are taking that

Reviewer 2: ```suggestion
Users that are familiar with Camel K and its "ease of use", then we are taking that
```

same user experience to everyone; without the need for a Kubernetes platform.

Having JBang installed, then you can easily run a Camel application from a terminal.

To quickly create a new basic Camel integration use the `init` command:

    camel init hello.java

And then you can run the file:

    camel run hello.java

We made it easy to run existing examples directly loaded from GitHub:

    camel run https://github.com/apache/camel-kamelets-examples/tree/main/jbang/hello-java

And we also made it easy to download the example to local disk, so you can edit the source and

> see live updates, using the `init` command.

Reviewer 2: ```suggestion
see live updates, using the `init` command:
```

    camel init https://github.com/apache/camel-kamelets-examples/tree/main/jbang/hello-java

And then you can run in developer mode that has live reloading on source changes (on save):

    camel run * --dev

We continue our effort on Camel JBang for the upcoming 3.18 LTS release.

There is a lot more that Camel JBang can do, so make sure to see the [Camel JBang documentation](/manual/camel-jbang.html).
There you can also find information how to get JBang installed, and after that install the camel app in JBang.

### Component Headers

> We completed the work of documenting every header every component supports.

Reviewer 3: ```suggestion
We completed the work of documenting every headers of all supported components.
```

This means you can see in the website documentation what headers a component consumer or producer

> uses, and its purpose.

Reviewer 3: ```suggestion
uses, and their purpose.
```

This information is also available for tooling. For example the Camel IDEA plugin supports this
and can do code assistance in your Camel routes.

### Resume from Offset

> TODO: Otavio anything to mention (pausable/resumable)

Reviewer 4: @davsclaus I'll try to complement this today afternoon.

### Java DSL in Camel K

The `camel-java-joor-dsl` used by Camel K, and Camel JBang, now supports compiling all Java sources
in the same compilation unit, meaning they have the same classloader visibility and can use
dependency injection among themselves. 

### Camel YAML DSL

The YAML parser now have precise source:line error reporting, meaning you can see exactly
where there is a problem.

> ### Load properties from valut/secrets cloud services

Reviewer 3: ```suggestion
### Load properties from vault/secrets cloud services
```

> Experimental support for resume operations was added to the following components in this release: camel-atom, camel-rss, camel-cassandraql and camel-couchbase. 

Reviewer 5: ```suggestion
Experimental support for resume operations was added to the following components in this release: `camel-atom`, `camel-rss`, `camel-cassandraql` and `camel-couchbase`. 
```

We added support for Azure key vault service.
See more details at the following [blog post](/blog/2022/03/secrets-properties-functions/)

### Camel Kafka

TODO: More Kafka ???

> We did other bug fixes, and we continue to make the camel-kafka component more robust. 1

Reviewer 4: @davsclaus I'll try to complement this today afternoon. We have a few fixes and changes affecting this component.

Reviewer 6: I'd like to mention that we support kafka transaction with https://issues.apache.org/jira/browse/CAMEL-15016

Author: @zhfeng 3.17 is being released today, do you have time to submit the kafka transaction text?

Reviewer 6: @davsclaus yeah, I will submit it. sorry for the delay!

Reviewer 6: @davsclaus I added it.

### Spring Boot

We have upgraded to the latest Spring Boot 2.6 release.

Added more unit tests to various starter JARs to increase the QA of this project.

## Upgrading

Make sure to read the [upgrade guide](/manual/camel-3x-upgrade-guide-3_17.html) if you are upgrading from a previous Camel version.

## Release Notes

> You can find more information about this release in the with a list of JIRA tickets resolved in the release: 

Reviewer 3: ```suggestion
You can find more information about this release in the list of JIRA tickets resolved in the release: 
```
